common/nns.go

```diff
@@ -0,0 +1,43 @@
+package common
+
+import (
+	"github.com/nspcc-dev/neo-go/pkg/interop"
+	"github.com/nspcc-dev/neo-go/pkg/interop/contract"
+	"github.com/nspcc-dev/neo-go/pkg/interop/convert"
+	"github.com/nspcc-dev/neo-go/pkg/interop/lib/address"
+	"github.com/nspcc-dev/neo-go/pkg/interop/native/management"
+	"github.com/nspcc-dev/neo-go/pkg/interop/native/std"
+)
+
+// ResolveContractHash resolves contract hash by its well-known NeoFS name.
+// Contract name should be lowercased, should not include `.neofs` TLD. Example
+// values: "netmap", "container", etc.
+// Relies on some NeoFS specifics:
+//  1. NNS contract should be deployed first (and should have `1` contract ID)
+//  2. It should be prefilled with contract hashes by their names (no
+//     capitalized chars; no `neofs` TLD)
+func ResolveContractHash(contractName string) interop.Hash160 {
+	// get NNS contract (it always has ID=1 in the NeoFS Sidechain)
+	nnsContract := management.GetContractByID(1)
+	if nnsContract == nil {
+		panic("missing NNS contract")
+	}
+
+	resResolve := contract.Call(nnsContract.Hash, "resolve", contract.ReadOnly,
+		contractName+".neofs", 16, // TXT
```

Comment:
How about `nns.TXT`?

Reply:
import cycle, unfortunately. that is how i found https://github.com/nspcc-dev/neo-go/issues/3202